include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -492,11 +492,11 @@ struct __usm_or_buffer_accessor
   public:
     // Buffer accessor
     __usm_or_buffer_accessor(sycl::handler& __cgh, sycl::buffer<_T, 1>* __sycl_buf)
-        : __acc(sycl::accessor(*__sycl_buf, __cgh, sycl::read_write, __dpl_sycl::__no_init{}))
+        : __acc(*__sycl_buf, __cgh, __dpl_sycl::__no_init{})
```

Comment:
This isn't a new bug from this PR, but I think that `__no_init` it is wrong to be using at all times here I think this is something that we need to specify each time and only when it is proper.
Well this is temporary memory it's often used to bridge between multiple kernels where the second kernel reads before it writes (if it does write). In this case `__no_init` is wrong I believe.

Reply:
Issue created: https://github.com/oneapi-src/oneDPL/issues/1914

Reply:
From https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_queue_interface,
Table 52. Properties supported by accessors
"This property is allowed only for accessors with access_mode::write or access_mode::read_write access modes. Attempting to construct an access_mode::read accessor with this property causes an exception with the errc::invalid error code to be thrown"

Reply:
@danhoeflinger for compatibility with the current code and based on documentation, I propose to set up `__dpl_sycl::__no_init{}` depends on used `_AccessMode` state.

Reply:
I think we should just fix it here (even though it grows beyond the original scope of the PR).  I don't think we should create a new system here to propagate the bug only to later replace it.  I can help advise where `no_init` is and isn't needed.
For `read`, its certainly not supposed to be `no_init`. For `write`, it should be `no_init`. For `read_write`, it will depend how it is used (written before read?).  A good clue is if it is the first kernel using it or not. Generally subsequent kernels should not be `no_init`.

Reply:
Implemented, please check.

Reply:
i think after the fix this looks good.  Thanks for adding this.